src/core/serialize/src/main/scala/params/DataFrameParam.scala

```diff
@@ -0,0 +1,29 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package org.apache.spark.ml.param
+
+import org.apache.spark.sql.{DataFrame, SparkSession}
+
+/** Param for DataFrame.  Needed as spark has explicit params for many different
+  * types but not DataFrame.
+  */
+class DataFrameParam(parent: Params, name: String, doc: String, isValid: DataFrame => Boolean)
+  extends ComplexParam[DataFrame](parent, name, doc, isValid) {
+
+  def this(parent: Params, name: String, doc: String) =
+    this(parent, name, doc, ParamValidators.alwaysTrue)
+
+  override def jsonEncode(value: DataFrame): String = {
```

Comment:
How big is this dataframe going to be? If its anything more than a few lines we should modify this to wrap and unwrap the dataframe in python

Reply:
the biggest it could be would be 10mil rows and 40k columns... so that's probably big...